flow/project.py

```diff
@@ -479,24 +469,491 @@ def complete(self, job):
+        :class:`dict`
+    :param options:
+        A string of options to append to the output of the object's call method.
+        This lets options like --num_passes to be given to a group.
+    :type options:
+        :class:`str`
+
+    ..code:: python
+        group = FlowProject.make_group(name='example_group')
+
+        @group.with_directives(nranks=4)
+        @FlowProject.operation
+        @directives(nranks=2, executable="python3")
+        def op1(job):
+            pass
+
+        @group
+        @FlowProject.operation
+        @directives(nranks=2, executable="python3")
+        def op1(job):
+            pass
+
+        # the directives used will be {'nranks': 4} for op1
+        # and {'nranks': 2, 'executable': 'python3'} for op2
+    """
+
+    MAX_LEN_ID = 100
+
+    def __init__(self, name, operations=None, operation_directives=None,
+                 options=""):
+        self.name = name
+        self.options = options
+        self.operations = dict() if operations is None else operations
+        if operation_directives is None:
+            self.operation_directives = dict()
+        else:
+            self.operation_directives = operation_directives
+
+    def _set_entrypoint_item(self, entrypoint, directives, key, default, job):
+        """Set a value (executable, path) for entrypoint in command.
+
+        Order of priority is the operation directives specified and
+        then the project specified value.
+        """
+        entrypoint[key] = directives.get(key, entrypoint.get(key, default))
+        if callable(entrypoint[key]):
+            entrypoint[key] = entrypoint[key](job)
+
+    def _determine_entrypoint(self, entrypoint, directives, job):
+        """Get the entrypoint for creating a JobOperation.
+
+        If path cannot be determined, then raise a RuntimeError since we do not
+        know where to point to.
+        """
+        entrypoint = entrypoint.copy()
+        self._set_entrypoint_item(entrypoint, directives, 'executable', sys.executable, job)
+        self._set_entrypoint_item(entrypoint, directives, 'path', None, job)
+        if entrypoint['path'] is None:
+            raise RuntimeError("Entrypoint path not set.")
+        else:
+            return "{} {}".format(entrypoint['executable'], entrypoint['path']).lstrip()
+
+    def _resolve_directives(self, name, defaults):
+        if name in self.operation_directives:
+            return deepcopy(self.operation_directives[name])
+        else:
+            return deepcopy(defaults.get(name, dict()))
+
+    def _submit_cmd(self, entrypoint, ignore_conditions, job=None):
+        entrypoint = self._determine_entrypoint(entrypoint, dict(), job)
+        cmd = "{} run -o {}".format(entrypoint, self.name)
+        cmd = cmd if job is None else cmd + ' -j {}'.format(job)
+        cmd = cmd if self.options is None else cmd + ' ' + self.options
+        return cmd.strip() + ' --ignore-conditions=' + str(ignore_conditions)
```

Comment:
Should we maybe only add the `--ignore-conditions` option if it is relevant? Otherwise this might create a lot of noise in our scripts.

Reply:
I was trying to leave the `ignore_conditions` logic in `__str__` to decouple the objects a bit more, but we could do a check for ` if cond_str == 'none'`

Reply:
Why not `if cond != CONDITIONS.NONE` ? (I'm not fully familiar with the conditions so please don't take that literally.

Reply:
We could do that, I was just attempting to make the logic simpler